shell/key-bindings.fish

```diff
@@ -59,27 +57,27 @@ function fzf_key_bindings
   function fzf-history-widget -d "Show command history"
     test -n "$FZF_TMUX_HEIGHT"; or set FZF_TMUX_HEIGHT 40%
     begin
-      set -l FISH_MAJOR (echo $version | cut -f1 -d.)
-      set -l FISH_MINOR (echo $version | cut -f2 -d.)
+      set -l FISH_MAJOR (string split '.' -- $version)[1]
+      set -l FISH_MINOR (string split '.' -- $version)[2]
 
       # merge history from other sessions before searching
-      if test -z "$fish_private_mode"
-        builtin history merge
-      end
+      test -z "$fish_private_mode"; and builtin history merge
 
       # history's -z flag is needed for multi-line support.
       # history's -z flag was added in fish 2.4.0, so don't use it for versions
       # before 2.4.0.
-      if [ "$FISH_MAJOR" -gt 2 -o \( "$FISH_MAJOR" -eq 2 -a "$FISH_MINOR" -ge 4 \) ];
-        if type -P perl > /dev/null 2>&1
-          set -lx FZF_DEFAULT_OPTS (__fzf_defaults "" "-n2..,.. --scheme=history --bind=ctrl-r:toggle-sort --wrap-sign '"\t"↳ ' --highlight-line $FZF_CTRL_R_OPTS +m")
-          set -lx FZF_DEFAULT_OPTS_FILE ''
-          builtin history -z --reverse | command perl -0 -pe 's/^/$.\t/g; s/\n/\n\t/gm' | eval (__fzfcmd) --tac --read0 --print0 -q '(commandline)' | command perl -pe 's/^\d*\t//' | read -lz result
+      if test "$FISH_MAJOR" -gt 2 -o \( "$FISH_MAJOR" -eq 2 -a "$FISH_MINOR" -ge 4 \)
+        set -lx FZF_DEFAULT_OPTS (__fzf_defaults "" "-n2..,.. --scheme=history --bind=ctrl-r:toggle-sort --wrap-sign '"\t"↳ ' --highlight-line $FZF_CTRL_R_OPTS +m")
+        set -lx FZF_DEFAULT_OPTS_FILE ''
+        if type -q perl
+          builtin history -z --reverse | command perl -0 -pe 's/^/$.\t/g; s/\n/\n\t/gm' | eval (__fzfcmd) --tac --read0 --print0 -q '(commandline)' | string replace -r '^\d*\t' '' | read -lz result
           and commandline -- $result
         else
-          set -lx FZF_DEFAULT_OPTS (__fzf_defaults "" "--scheme=history --bind=ctrl-r:toggle-sort --wrap-sign '"\t"↳ ' --highlight-line $FZF_CTRL_R_OPTS +m")
-          set -lx FZF_DEFAULT_OPTS_FILE ''
-          builtin history -z | eval (__fzfcmd) --read0 --print0 -q '(commandline)' | read -lz result
+          set -l line 0
+          for i in (builtin history -z --reverse | string split0)
+            set line (math $line + 1)
+            string escape -n -- $line\t$i
+          end | string join0 | string replace -a '\n' '\n\t' | string unescape -n | eval (__fzfcmd) --tac --read0 --print0 -q '(commandline)' | string replace -r '^\d*\t' '' | read -lz result
```

Comment:
Have you compared the performance of the two versions? If the native fish script is as performant as the Perl version, we can just remove that and simplify the code.

Reply:
Yes, I have. On large history files with thousands of items, there is a slight delay (with perl it's instant). So, I think it's better to keep it (I forgot to write the reasoning in the commit message, sorry).

```
> time ./bench-hist-builtin.fish

________________________________________________________
Executed in  608.92 millis    fish           external
   usr time  506.29 millis    2.51 millis  503.79 millis
   sys time  138.58 millis    0.37 millis  138.22 millis


> time ./bench-hist-perl.fish

________________________________________________________
Executed in   75.61 millis    fish           external
   usr time   73.59 millis    2.85 millis   70.74 millis
   sys time   12.99 millis    0.00 millis   12.99 millis

```